### PR TITLE
Drop box provider 

### PR DESCRIPTION
**Pull request by SH2015**
Some changes might not be required in xproj files , but I needed to build the application

### CODE REVIEW

File: src/AspNet.Security.OAuth.DeviantArt/DeviantArtAuthenticationExtensions.cs [hunk 1]
```diff
@@ -6,27 +6,23 @@
...
     public static class DeviantArtAuthenticationExtensions {
-        public static IServiceCollection ConfigureDeviantArtAuthentication(
-            [NotNull] this IServiceCollection services,
-            [NotNull] Action<DeviantArtAuthenticationOptions> configuration) {
-            return services.Configure(configuration);
-        }
-
-        public static IApplicationBuilder UseDeviantArtAuthentication([NotNull] this IApplicationBuilder app) {
-            return app.UseMiddleware<DeviantArtAuthenticationMiddleware>();
+        public static IApplicationBuilder UseDeviantArtAuthentication(
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.DeviantArt/DeviantArtAuthenticationExtensions.cs]: These changes seem unrelated. Make sure to run `git fetch upstream` + `git rebase upstream/dev` + `git push -f` to rebase your branch on top of the latest dev state :+1:

File: src/AspNet.Security.OAuth.Dropbox/DropboxAuthenticationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            response.EnsureSuccessStatusCode();
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            identity.AddOptionalClaim(ClaimTypes.NameIdentifier, DropboxAuthenticationHelper.GetIdentifier(payload), Options.ClaimsIssuer).AddOptionalClaim(ClaimTypes.Name, DropboxAuthenticationHelper.GetDisplayName(payload), Options.ClaimsIssuer).AddOptionalClaim(ClaimTypes.Email, DropboxAuthenticationHelper.GetEmail(payload), Options.ClaimsIssuer);
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Dropbox/DropboxAuthenticationHandler.cs]: This line is a bit... long? :smile: 
Please add a few line breaks.

File: src/AspNet.Security.OAuth.Dropbox/DropboxAuthenticationHelper.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    /// <summary>
+    /// Contains static methods that allow to extract user's information from a <see cref="JObject"/>
+    /// instance retrieved from Dropbox after a successful authentication process.
+    /// </summary>
+    public static class DropboxAuthenticationHelper {
+        /// <summary>
+        /// Gets the identifier corresponding to the authenticated user.
+        /// </summary>
+        public static string GetIdentifier([NotNull] JObject user) => user.Value<string>("id");
+        public static string GetDisplayName ([NotNull] JObject user) => user.Value<string>("display_name");
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Dropbox/DropboxAuthenticationHelper.cs]: Please add comments on these methods.

**Comment by PinpointTownes** [comment]
Thanks for your contribution! :+1: 

Minor my few remarks (make sure to run `git fetch upstream` + `git rebase upstream/dev` + `git push -f` to fix the merge issues), it looks fine.

/cc @jerriep

**Comment by jerriep** [comment]
@PinpointTownes  Yeah, I also noticed all the extra changes to non-relevant files but waited for you to review as you would be better able to comment on how to fix it - as you did :) Nothing else from me.

File: src/AspNet.Security.OAuth.Dropbox/DropboxAuthenticationHelper.cs (hunk 3 above)
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Dropbox/DropboxAuthenticationHelper.cs]: Oh, and remove the space before the opening parenthesis :smile:

**Comment by SH2015** [comment]
@PinpointTownes  : het Kevin , not sure to which branch you want the rebase to be, is it  to my fork branch or to aspnet-contrib/AspNet.Security.OAuth.Providers/dev branch ?

**Comment by PinpointTownes** [comment]
Yeah, you must rebase your fork branch on top of aspnet-contrib/AspNet.Security.OAuth.Providers/dev (probably referred as "upstream") using the commands I mentioned :+1:
